Gems/DebugDraw/Code/CMakeLists.txt

```diff
@@ -59,7 +59,10 @@ if(PAL_TRAIT_BUILD_HOST_TOOLS)
                 DEBUGDRAW_GEM_EDITOR=1
         BUILD_DEPENDENCIES
             PRIVATE
-                Gem::DebugDraw.Static
+                AZ::AtomCore
+                Gem::Atom_RPI.Public
+                Gem::Atom_Bootstrap.Headers
+                Legacy::CryCommon
```

Comment:
It looks like here that the [debugdraw_editor_files.cmake](https://github.com/o3de/o3de/blob/development/Gems/DebugDraw/Code/debugdraw_editor_files.cmake) contains a superset of the [debugdraw_files.cmake](https://github.com/o3de/o3de/blob/development/Gems/DebugDraw/Code/debugdraw_files.cmake) from the Static library target.

We could remove the same files contained in the `debugdraw_editor_files.cmake` and still link the `Gem::DebugDraw.Static`

Reply:
https://github.com/o3de/o3de/blob/602b668c35203c7c3db590c0a8ae126de8cac306/Gems/DebugDraw/Code/CMakeLists.txt#L59

When building `DebugDraw.Editor`, it needs a macro `DEBUGDRAW_GEM_EDITOR` to enable some functions that are not used in runtime.
That's why it doesn't work to remove the same files contained in the `debugdraw_editor_files.cmake` and still link the `Gem::DebugDraw.Static`.

Reply:
Thanks for the explanation. It is not ideal that we reuse the same files for both Editor and non-Editor using `#if`, but it is fine.